backend/dal/Services/InsuranceService.cs

```diff
@@ -0,0 +1,95 @@
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Security.Claims;
+using MapsterMapper;
+using Microsoft.Extensions.Logging;
+using Pims.Dal.Entities;
+using Pims.Dal.Helpers.Extensions;
+using Pims.Dal.Security;
+
+namespace Pims.Dal.Services
+{
+    /// <summary>
+    /// InsuranceService class, provides a service layer to interact with insurances within the datasource.
+    /// </summary>
+    public class InsuranceService : BaseService<PimsInsurance>, IInsuranceService
+    {
+        #region Constructors
+        /// <summary>
+        /// Creates a new instance of a InsuranceService, and initializes it with the specified arguments.
+        /// </summary>
+        /// <param name="dbContext"></param>
+        /// <param name="user"></param>
+        /// <param name="service"></param>
+        /// <param name="logger"></param>
+        public InsuranceService(PimsContext dbContext, ClaimsPrincipal user, IPimsService service, ILogger<InsuranceService> logger, IMapper mapper) : base(dbContext, user, service, logger, mapper) { }
+        #endregion
+
+        #region Methods
+        public PimsInsurance Get(long id)
+        {
+            this.User.ThrowIfNotAuthorized(Permissions.LeaseView);
+            return this.Context.PimsInsurances
+                .Where(i => i.InsuranceId == id)
+                .FirstOrDefault() ?? throw new KeyNotFoundException();
```

Comment:
nit, I think this can be shortened to FirstOrDefault(i => i.InsuranceId == id).

Could also consider making this non-tracking. That way if someone calls get, inadvertently updates the entity while doing other work, those changes wouldn't be saved.